src/test/java/apoc/export/cypher/ExportCypherEnterpriseFeaturesTest.java

```diff
@@ -0,0 +1,136 @@
+                    "CREATE (b:Person {name: 'Matt', surname: 'Jackson'}) " +
+                    "CREATE (c:Person {name: 'Jenny', surname: 'White'}) " +
+                    "CREATE (d:Person {name: 'Susan', surname: 'Brown'}) " +
+                    "CREATE (e:Person {name: 'Tom', surname: 'Taylor'})" +
+                    "CREATE (a)-[:KNOWS]->(b);");
+            tx.success();
+            return null;
+        });
+    }
+
+    @AfterClass
+    public static void afterAll() {
+        if (neo4jContainer != null) {
+            session.close();
+            driver.close();
+            neo4jContainer.close();
+        }
+        cleanBuild();
+    }
+
+    @Test
+    public void testExportWithCompoundConstraintCypherShell() throws Exception {
+        String fileName = "testCypherShellWithCompoundConstraint.cypher";
+        File output = getFile(fileName);
+        testCall(session, "CALL apoc.export.cypher.all(null,{config})",
+                map("file", getFilePath(output), "config", Util.map("format", "cypher-shell"), "stream", isTravis()), (r) -> {
+                    assertExportStatement(EXPECTED_CYPHER_SHELL_WITH_COMPOUND_CONSTRAINT, output, r);
+                });
+    }
+
+    @Test
+    public void testExportWithCompoundConstraintPlain() throws Exception {
+        File output = getFile("testPlainFormatWithCompoundConstraint.cypher");
+        testCall(session, "CALL apoc.export.cypher.all({file},{config});",
+                map("file", getFilePath(output), "config", Util.map("format", "plain"), "stream", isTravis()),
+                (r) -> assertExportStatement(EXPECTED_PLAIN_FORMAT_WITH_COMPOUND_CONSTRAINT, output, r));
+    }
+
+    @Test
+    public void testExportWithCompoundConstraintNeo4jShell() throws Exception {
+        File output = getFile("testNeo4jShellWithCompoundConstraint.cypher");
+        testCall(session, "CALL apoc.export.cypher.all({file},{config});",
+                map("file", getFilePath(output), "config", Util.map("format", "neo4j-shell"), "stream", isTravis()),
+                (r) -> assertExportStatement(EXPECTED_NEO4J_SHELL_WITH_COMPOUND_CONSTRAINT, output, r));
+    }
+
+    private void assertExportStatement(String expectedStatement, File output, Map<String, Object> result) {
+        assertEquals(expectedStatement, isTravis() ? result.get("cypherStatements") : readFile(output));
+    }
+
+    private static String readFile(File output) {
+        try {
+            return new Scanner(output).useDelimiter("\\Z").next() + String.format("%n");
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private String getFilePath(File output) { // TODO on Travis we get FileNotFoundException with "Permission Denied" so we test the Cypher from the map (investigate why)
+        return isTravis() ? "null" : PREFIX + output.getPath().replace("target/", "");
+    }
+
+    private File getFile(String fileName) {
```

Comment:
what is this achieving?
the method name is not really clear

it actually writes a  new, empty file to the location?

What about file.exits(), canWrite(), isDirectory() etc?

Reply:
They are tries that I made when I realized that on Travis I had permissions issue, I forgot to delete them sorry.